### PR TITLE
chore: update to latest `gen` module

### PR DESCRIPTION
**Pull request by DmitriyMV**
Update and remove stuff, such as:
- Run rekres
- Bump Go version to 1.21.0
- Remove deprecated `channel.RecvWithContext`.
- Use `ensure.Value` where possible.
- Use functions from `check` package for table-driven tests.
- Use `must.Value` instead of custom `must` function.
- Use std `slices` where possible.

This is an example for https://github.com/siderolabs/gen/pull/14

### CODE REVIEW

**Review by smira — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -1,11 +1,13 @@
 module github.com/cosi-project/runtime
 
-go 1.21
+go 1.21.0
 
 // forked yaml that introduces RawYAML interface that can be used to provide YAML encoder bytes
 // which are then encoded as a valid YAML block with proper indentiation
 replace gopkg.in/yaml.v3 => github.com/unix4ever/yaml v0.0.0-20220527175918-f17b0f05cf2c
 
+replace github.com/siderolabs/gen => github.com/DmitriyMV/gen v0.0.0-20230822105703-36a3ae312ce0
```
Comment (frezbo) [inline go.mod:9]: this missed?

**Review by frezbo — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (DmitriyMV) [reply go.mod:9]: `Gen` not yet merged.

Comment (DmitriyMV) [reply go.mod]: Removed!

**Pull request merged**